### PR TITLE
Update FAQ to reflect current changes in Open API

### PR DESCRIPTION
**Pull request by artemkovalyov**
## What Has Changed?

Explain what you have changed.

## Manual Checks?

- [x] Text adheres to the style guide (`vale docs/`)
  - Every sentence is on it's own line
  - Headings use [title capitalization](https://capitalizemytitle.com/style/AP/#) (applies also to the sidebar and title of a document)
  - You followed [naming center](https://www.sapbrandtools.com/naming-center/#/dashboard) guidelines when referring to SAP products (e.g. SAP S/4HANA)
- [x] You checked your spelling and grammar (consider using Grammarly, see `CONTRIBUTING.md`)
- [x] You formatted all changed files with prettier (`npm run prettier`)
- [x] You tested if the documentation still builds (`npm run build`)
- [x] You verified all new and changed links still work (changing the `id` or name of a file can break links)

### CODE REVIEW

**Comment by sacnl** [comment]
With this change, how will we direct people that just look for "REST support" and may not know or care about Open API?

**Comment by artemkovalyov** [comment]
> With this change, how will we direct / reach people that just look for "REST support" and may not know or care about Open API?

I'm confused a little. OData is mostly REST, Open API is REST, HTTP is REST, and so on. I think the search term is valid still, so I'll check with the Google search console and SWA. In my head REST would be smth like calling RESTful API endpoints via our connectivity layer? I thought that people use smth like Spring REST template for that?

On the Cloud SDK side, REST would be mostly equal to Open API. I was surprised we call it REST on the API Hub as well as it's Open API 2 or 3.

In the end, we'll keep REST as a keyword somewhere, but in relation to type-safe clients, I see Open API as a more relevant term.

**Comment by sacnl** [comment]
You could also additionally mention "plain REST" or similar (i.e., not additionally following OData protocol) to establish the connection to what is mentioned elsewhere, like on API Hub. Some readers will come with that in mind and look for REST support by the SDK (to further complicate it, OData services can also be represented as OpenAPI - and are on API Hub, if I'm not mistaken).

File: docs/java/faq.mdx [hunk 1]
```diff
@@ -79,16 +79,15 @@ These errors are not always caused by flaws in the SAP Cloud SDK as we often see
 Some of them even have known flaws for which we have workarounds.
 If you can't solve your issue via debugging and experimenting, please, report your incident via [Stack Overflow](https://stackoverflow.com/questions/tagged/sap-cloud-sdk) or our GitHub repositories.
 
-## REST Related Questions
+## Open API Related Questions
 
-### Do You Support REST Client Libraries?
+### Do You Support Open API Client Libraries?
```
Comment (artemkovalyov) [inline docs/java/faq.mdx:84]:
```suggestion
### Do You Support RESTful Services?
```

File: docs/java/faq.mdx [hunk 2]
```diff
@@ -79,16 +79,15 @@ These errors are not always caused by flaws in the SAP Cloud SDK as we often see
...
 
-Yes, we do.
-We do not release a public REST client generator yet.
-We have a set of libraries supplied together with SAP Cloud SDK for Java.
-Some of them are available only for SAP internal use.
-Others, like the Workflow service on Cloud Foundry, are released publicly.
-Check our [REST capabilities](features/rest/overview) and let us know if you need a library for an SAP service that you use and know to be providing REST API.
+Yes, we do for Open API 3.0 specifications.
```
Comment (artemkovalyov) [inline docs/java/faq.mdx:86]:
```suggestion
Yes, we do support [OpenAPI 2.0 also known as Swagger](https://swagger.io/specification/v2/), and  [OpenAPI 3.0](https://swagger.io/specification/) specifications. The latter is preferred.
```

File: docs/java/faq.mdx [hunk 3]
```diff
@@ -79,16 +79,15 @@ These errors are not always caused by flaws in the SAP Cloud SDK as we often see
...
+Find documentation how to generate a [type-safe Open API client here](features/rest/generate-rest-client.mdx).
+We also release pre-generated libraries in colaboration with service providers who release restful Open API services.
```
Comment (artemkovalyov) [inline docs/java/faq.mdx:88]:
```suggestion
We also release pre-generated libraries in collaboration with service providers who release RESTful Open API services.
```

**Review by artemkovalyov — COMMENTED** [review]
!

**Comment by artemkovalyov** [comment]
@sacnl, take another look

File: docs/java/faq.mdx [hunk 4]
```diff
@@ -79,16 +79,15 @@ These errors are not always caused by flaws in the SAP Cloud SDK as we often see
...
-### Do You Support REST Client Libraries?
+### Do You Support RESTful Services?
 
-Yes, we do.
-We do not release a public REST client generator yet.
-We have a set of libraries supplied together with SAP Cloud SDK for Java.
-Some of them are available only for SAP internal use.
-Others, like the Workflow service on Cloud Foundry, are released publicly.
-Check our [REST capabilities](features/rest/overview) and let us know if you need a library for an SAP service that you use and know to be providing REST API.
+Yes, we do support [OpenAPI 2.0 also known as Swagger](https://swagger.io/specification/v2/), and  [OpenAPI 3.0](https://swagger.io/specification/) specifications. The latter is preferred.
```
Comment (artemkovalyov) [inline docs/java/faq.mdx:86]:
```suggestion
Yes, we do support [OpenAPI 2.0 also known as Swagger](https://swagger.io/specification/v2/), and  [OpenAPI 3.0](https://swagger.io/specification/) specifications.
The latter is preferred.
```

**Review by artemkovalyov — COMMENTED** [review]
!!

**Comment by artemkovalyov** [comment]
@Emdee89 @KavithaSiva , I'd happy with your review.

**Comment by sacnl** [comment]
> @sacnl, take another look

Thanks, looks good. I leave the final review to the others.

**Review by Emdee89 — APPROVED** [review]

**Pull request merged**